jenkins/DataLoader/OpensearchBackup.jenkinsfile

```diff
@@ -0,0 +1,85 @@
+import groovy.json.JsonOutput
+
+pipeline {
+  agent {
+    node {
+      label 'slave-ncias-d2957-c'
+    }
+  }
+	parameters {
+    extendedChoice( 
+        name: 'Environment', 
+        defaultValue: 'dev', 
+        description: 'Choose the environment to build', 
+        type: 'PT_SINGLE_SELECT',
+        value: 'dev,qa,stage,prod' )
+    string(defaultValue: "ctdc",
+        description: 'snapshot repository name',
+        name: 'PROJECT_NAME')
+    string(defaultValue: "ctdc",
+        description: 'snapshot repository name',
+        name: 'SNAPSHOT_REPO')
+    string(defaultValue: "v_1_0",
+        description: 'Snapshot name used for restore operation',
+        name: 'SNAPSHOT_VALUE')
+    string(defaultValue: "crdc-stage-ctdc-opensearch-snapshot-bucket",
+        description: 'S3 bucket which has the snapshot to restore',
+        name: 'S3_BUCKET')
+    string(defaultValue: "es-backup",
+        description: 'sub folder in S3 bucket where the snapshot is present',
+        name: 'BASE_PATH')
+  }
+  options {
+  	ansiColor('xterm')
+  }
+  tools {
+  	maven 'Default' 
+    jdk 'Default' 
+  }
```

Comment:
can there be an env block added where we will define variables like the project, repos, branches and other project specific values so that we can make this into a template - in general with template files the only sections that will need editing to adapt it to a specific project will be the parameters (for repo URLs) and the environment variables